e2e/test/scenarios/dashboard-cards/click-behavior.cy.spec.js

```diff
@@ -831,6 +931,69 @@ describe("scenarios > dashboard > dashboard cards > click behavior", () => {
       })();
     });
 
+    it("should allow setting dashboard tab with parameter for a column", () => {
+      cy.createQuestion(TARGET_QUESTION);
+      cy.createDashboard(
+        {
+          ...TARGET_DASHBOARD,
+          parameters: [DASHBOARD_FILTER_TEXT, DASHBOARD_FILTER_TIME],
+        },
+        {
+          wrapId: true,
+          idAlias: "targetDashboardId",
+        },
+      );
+      cy.get("@targetDashboardId").then(targetDashboardId => {
+        cy.request("PUT", `/api/dashboard/${targetDashboardId}/cards`, {
+          cards: [],
+          tabs: [FIRST_TAB, SECOND_TAB, THIRD_TAB],
+        });
+      });
+      cy.createQuestionAndDashboard({ questionDetails }).then(
+        ({ body: card }) => {
+          visitDashboard(card.dashboard_id);
+        },
+      );
+
+      editDashboard();
+
+      getDashboardCard().realHover().icon("click").click();
+      cy.get("aside").findByText(COUNT_COLUMN_NAME).click();
+      addDashboardDestination();
+      cy.get("aside")
+        .findByLabelText("Select a dashboard tab")
+        .should("have.value", FIRST_TAB.name)
+        .click();
+      cy.findByRole("listbox").findByText(SECOND_TAB.name).click();
+      cy.get("aside").findByText("No available targets").should("not.exist");
+      addTextParameter();
+
+      cy.icon("chevronleft").click();
+
+      getCountToDashboardMapping().should("exist");
+      getDashboardCard()
+        .button()
+        .should("have.text", "1 column has custom behavior");
+
+      cy.get("aside").button("Done").click();
+      saveDashboard();
+
+      getTableCell(COLUMN_INDEX.COUNT)
+        .should("have.text", String(POINT_COUNT))
+        .click();
+      cy.findAllByTestId("field-set")
+        .should("have.length", 2)
+        .should("contain.text", POINT_COUNT);
+      cy.get("@targetDashboardId").then(targetDashboardId => {
+        cy.location().should(({ pathname, search }) => {
+          expect(pathname).to.equal(`/dashboard/${targetDashboardId}`);
+          expect(search).to.equal(
+            `?tab=${SECOND_TAB_SLUG}&${DASHBOARD_FILTER_TEXT.slug}=${POINT_COUNT}&${DASHBOARD_FILTER_TIME.slug}=`,
+          );
+        });
+      });
+    });
+
```

Comment:
I'd like to see tests both for missing `tabId` and when the tab was removed from the target dashboard

Reply:
> I'd like to see tests both for missing `tabId`

Added a test case for missing tabId: 58406528cd3088eb8ebb0ea4a86704bd573e92c2
Please notice mentions of both migration functions in comments in that commit.

> and when the tab was removed from the target dashboard

This already exists here: https://github.com/metabase/metabase/pull/35311/files#diff-3ca928da028c386c159c907bf8f40bee1d8e4bc52150e9ec46a5d0aca0acca29R353-R386